### PR TITLE
Add missing urls and examples for Condvar docs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

File: src/libstd/sync/condvar.rs [hunk 1]
```diff
@@ -18,12 +18,52 @@ use time::Duration;
...
+///
+/// [`wait_timeout`]: struct.Condvar.html#method.wait_timeout
 #[derive(Debug, PartialEq, Eq, Copy, Clone)]
 #[stable(feature = "wait_timeout", since = "1.5.0")]
 pub struct WaitTimeoutResult(bool);
 
 impl WaitTimeoutResult {
     /// Returns whether the wait was known to have timed out.
+    ///
+    /// # Examples
```
Comment (frewsxcv) [inline src/libstd/sync/condvar.rs]: There's a lot going on in the examples in this pull request, and I understand they probably can't get much smaller, but is it possible you could add a short sentence before each example giving an overview of what's going on?

In this particular example, maybe: "Using a CondVar to notify the main thread when when a child thread has started:"

Comment (GuillaumeGomez) [inline src/libstd/sync/condvar.rs]: I wonder: is a summary enough or wouldn't it be better to just add more comments?

Comment (frewsxcv) [inline src/libstd/sync/condvar.rs]: Up to you, though I don't think it would hurt to have both

Comment (GuillaumeGomez) [inline src/libstd/sync/condvar.rs]: Let's do both then!

**Comment by GuillaumeGomez** [comment]
So apparently, the `WaitTimeoutResult` returned from `wait_timeout` returns always true on `timed_out` function if it happens once. Not sure if this is expected or not...

**Comment by GuillaumeGomez** [comment]
cc @frewsxcv

**Pull request merged**